index.js

```diff
@@ -124,8 +124,15 @@ export default class Barcode extends PureComponent {
 
   // encode() handles the Encoder call and builds the binary string to be rendered
   encode(text, Encoder, options) {
-    // Ensure that text is a string
-    text = '' + text;
+    // If text is not a non-empty string, throw error.
+    if (typeof text !== "string" || text.length === 0) {
+      if (this.props.onError) {
+        this.props.onError(new Error('Barcode value must be a non-empty string'));
+        return;
+      } else {
```

Comment:
I think the else if unnecessary if there is a return statement inside the previous if statement

Reply:
I blindly copied code from other places. I just updated the code removing unnecessary `else`.

Please help me review it again @wonsikin. Thank you.